File: GuessingGame.cs
using System;

public class GuessingGame
{
    public static void Main(string[] args)
    {
        string playAgain = "yes";

        while (playAgain.ToLower() == "yes")
        {
            // Step 3: Generate a random magic number
            Random random = new Random();
            int magicNumber = random.Next(1, 101);
            int guess = -1;
            int guessCount = 0;

            Console.WriteLine("Welcome to the Guessing Game!");

            // Step 2: Loop until the guess is correct
            while (guess != magicNumber)
            {
                Console.Write("What is your guess? ");
                string input = Console.ReadLine();

                if (int.TryParse(input, out guess))
                {
                    guessCount++;

                    // Step 1: Determine if the guess is higher or lower
                    if (guess < magicNumber)
                    {
                        Console.WriteLine("Higher");
                    }
                    else if (guess > magicNumber)
                    {
                        Console.WriteLine("Lower");
                    }
                    else
                    {
                        Console.WriteLine("You guessed it!");
                    }
                }
                else
                {
                    Console.WriteLine("Invalid input. Please enter a number.");
                }
            }

            // Step 4: Inform the user of the number of guesses
            Console.WriteLine($"You guessed it in {guessCount} tries.");

            // Step 5: Ask if the user wants to play again
            Console.Write("Do you want to play again? (yes/no) ");
            playAgain = Console.ReadLine();
        }

        Console.WriteLine("Thanks for playing!");
    }
}

File: guessing_game.py
#I have included in my game an option that it gives the opponent an option to retry every time a guess is wrong

import random

def number_guessing_game():
    """
    A simple number guessing game where the user tries to guess a random number.
    """

    # Generate a random number between 1 and 100 (inclusive)
    secret_number = random.randint(1, 100)

    # Initialize the number of guesses
    guesses = 0

    print("Welcome to the Number Guessing Game!")
    print("I'm thinking of a number between 1 and 100.")

    while True:
        try:
            # Get the user's guess
            guess = int(input("Enter your guess: "))
            guesses += 1 #increment guess count

            # Check if the guess is correct
            if guess == secret_number:
                print(f"Congratulations! You guessed the number in {guesses} guesses.")
                break #exit loop if guess is correct
            # Provide hints if the guess is too high or too low
            elif guess < secret_number:
                print("Too low. Try again.")
            else:
                print("Too high. Try again.")

        except ValueError:
            # Handle invalid input (non-integer)
            print("Invalid input. Please enter a number.")

if __name__ == "__main__":
    number_guessing_game() #start the game